### PR TITLE
support signTransaction and javascript development with bower

### PR DESCRIPTION
**Pull request by denisgranha**
I needed these changes in order to develop a new web3 provider for Ledger Wallet, the U2F (Universal 2nd Factor) API it's written in Javascript and only works in javascript.

Also, update web3 to last version 0.18.4 in order to support signTransaction function.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ function Web3ProviderEngine(opts) {
 Web3ProviderEngine.prototype.start = function(){
   const self = this
   // start block polling
-  self._blockTracker.start()
+  // self._blockTracker.start()
```
Comment (flyswatter) [inline index.js]: Why have you stopped blockTracker starting on start?

File: index.js [hunk 2]
```diff
@@ -67,7 +67,7 @@ Web3ProviderEngine.prototype.send = function(payload){
 
 Web3ProviderEngine.prototype.sendAsync = function(payload, cb){
   const self = this
-  self._ready.await(function(){
+  // self._ready.await(function(){
```
Comment (flyswatter) [inline index.js]: Why have you commented out this ready callback?

File: index.js [hunk 3]
```diff
@@ -168,26 +168,9 @@ Web3ProviderEngine.prototype._inspectResponseForNewBlock = function(payload, res
...
-  // our blockTag-rewriting mechanism
-  if (-1 === self.currentBlock.number.compare(blockNumber)) {
-    console.log('_inspectResponseForNewBlock start')
-    self._blockTracker._performSync().then(() => {
-      console.log('_inspectResponseForNewBlock end')
-      cb(null, resultObj)
-    })
-  } else {
-    cb(null, resultObj)
-  }
```
Comment (flyswatter) [inline index.js]: Why is this block of code removed?  This doesn't have to do with either adding bower support or adding `signTransaction` support.

**Comment by denisgranha** [comment]
Hello @flyswatter, sorry there were commits on this PR not intented to be here, just in my local fork.
The stop of block tracker and removed code, it's because we need an offline provider, but I didn't wanted to include that on the PR

**Comment by denisgranha** [comment]
I will create another PR with only required commit later today

**Comment by flyswatter** [comment]
Ok, closing this for now then, thanks!

**Pull request closed without merging**